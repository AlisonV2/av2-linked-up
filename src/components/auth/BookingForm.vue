<template>
  <form class="auth-form" @submit.prevent="setBooking">
    <div class="row">
      <div class="col-12 text-center mb-4">
        <h2 class="title">Book a Stand</h2>
      </div>
    </div>
    <div class="form-floating mb-3">
      <input
        type="email"
        class="form-control"
        placeholder="Please confirm your email"
        v-model="email"
        required
      />
      <label>Please confirm your email</label>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button>Confirm</app-button>
    </div>
    <slot />
  </form>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports BookingForm
 * @type {Component}
 * @requires Sentry
 * @vue-data {number} emails - v-model
 * @vue-event resetPassword
 */
export default {
  props: ['id'],
  data() {
    return {
      email: '',
      error: null,
    };
  },
  methods: {
    setBooking() {
      const data = {
        eventId: this.id,
        email: this.email,
      }
      this.$store
        .dispatch('setBooking', data)
        .then(() => {
          this.$router.push({ name: 'AdminEvents' });
        })
        .catch((err) => {
          this.error = err.message;
          Sentry.captureException(err);
          return;
        });
    },
  },
};
</script>

<style lang="scss" scoped>
.title {
  color: $dark;
  font-size: 1.8rem;
  margin-bottom: 1.5rem;
}
</style>
