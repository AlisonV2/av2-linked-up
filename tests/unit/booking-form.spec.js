import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import BookingForm from '@/components/auth/BookingForm.vue';
import AppTitle from '@/components/app/Title.vue';
import AppButton from '@/components/app/Button.vue';

let actions = {
  setBooking: jest.fn(),
};

const store = createStore({
  actions,
});

const $router = {
  push: jest.fn(),
};

const $route = {
  name: 'AdminEvents',
};

const config = {
  data() {
    return {
      tickets: 2,
      error: null,
      id: '1'
    };
  },
  global: {
    plugins: [store],
    mocks: {
      $router,
      $route,
    },
    components: {
      'app-title': AppTitle,
      'app-button': AppButton,
    },
  },
};

/**
 * @module BookingFormTest
 */
describe('BookingForm.vue', () => {
  it('BookingForm Snapshot', () => {
    const component = shallowMount(BookingForm, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Dispatches setBooking action on submit', async () => {
    const component = shallowMount(BookingForm, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect(actions.setBooking).toHaveBeenCalled;
  });

  it('Expects router push to have been called with AdminEvents params', async () => {
    const component = shallowMount(BookingForm, config);
    component.find('form').trigger('submit.prevent');
    await component.vm.$nextTick();
    expect($router.push).toHaveBeenCalled;
  });
});
