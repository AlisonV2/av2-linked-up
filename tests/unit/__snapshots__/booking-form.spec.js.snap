// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingForm.vue BookingForm Snapshot 1`] = `
<form
  class="auth-form"
>
  <div
    class="row"
  >
    <div
      class="col-12 text-center mb-4"
    >
      <h2
        class="title"
      >
        Book a Stand
      </h2>
    </div>
  </div>
  <div
    class="form-floating mb-3"
  >
    <input
      class="form-control"
      placeholder="Please confirm your email"
      required=""
      type="email"
    />
    <label>
      Please confirm your email
    </label>
  </div>
  <div
    class="error mb-3 mt-3"
  />
  <div
    class="btn-center"
  >
    <app-button-stub />
  </div>
  
  
</form>
`;
